lib/LedgerSMB/Scripts/payment.pm

```diff
@@ -1166,9 +1168,15 @@ sub payment2 {
         notes => $request->{notes},
         overpayment         => \@overpayment,
         overpayment_account => \@overpayment_account,
-        overpayment_subtotal => LedgerSMB::PGNumber->new((sum map { $_->{amount} } @overpayment) // 0)->to_output(money => 1),
-        payment_total => LedgerSMB::PGNumber->new((sum map { $_->{amount} } @overpayment)
-            + (sum map { $_->{topay} } @invoice_data))->to_output(money => 1),
+        overpayment_subtotal => (
+              (sum map {LedgerSMB::PGNumber->from_input($_->{amount} || 0)} @overpayment)
+             + LedgerSMB::PGNumber->from_input(0) # never end up with undef
```

Comment:
Did you explicitly select `||` over `//`? The original code had the latter variant.